### ISSUE TITLE
Socket IO Client permanently connecting and disconnecting to Gateway

### ISSUE THREAD

**Issue by emmanuel-D** [nestjs/nest#8721]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

I created a Socket-IO Gateway by following the official Docs.
But the issue that I currently have is that, when a Client is connecting to the Gateway, then few seconds after that it got disconnected: And that many times.

Here are some Logs of that behaviour:
<img width="922" alt="Screenshot 2021-11-29 at 21 01 59" src="https://user-images.githubusercontent.com/37305687/143934945-0e75c0e0-27d6-4428-82a4-b2c1dd9deef3.png">

I also find similar issues a possible solution would have been to change the `socket-io-client` Version.

I alreay tried this, but it's still not working.

### Minimum reproduction code

https://github.com/nestjs/nest/tree/master/sample/02-gateways

### Expected behavior

The Client-Connection should be maintained and keep alive.
It should not going disconnected automatically.

### Package

- [ ] I don't know. Or some 3rd-party package
- [ ] <code>@nestjs/common</code>
- [ ] <code>@nestjs/core</code>
- [ ] <code>@nestjs/microservices</code>
- [ ] <code>@nestjs/platform-express</code>
- [ ] <code>@nestjs/platform-fastify</code>
- [X] <code>@nestjs/platform-socket.io</code>
- [X] <code>@nestjs/platform-ws</code>
- [ ] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [ ] Other (see below)

### NestJS version

8.2.3

### Packages versions

````
 "dependencies": {
    "@nestjs/common": "^8.2.3",
    "@nestjs/config": "^1.1.0",
    "@nestjs/core": "^8.2.3",
    "@nestjs/event-emitter": "^1.0.0",
    "@nestjs/jwt": "^8.0.0",
    "@nestjs/passport": "^8.0.1",
    "@nestjs/platform-express": "^8.2.3",
    "@nestjs/platform-socket.io": "^8.2.3",
    "@nestjs/schedule": "^1.0.2",
    "@nestjs/swagger": "^5.1.5",
    "@nestjs/typeorm": "^8.0.2",
    "@nestjs/websockets": "^8.2.3",
    "agora-access-token": "^2.0.4",
    "aws-sdk": "^2.858.0",
    "axios": "^0.21.1",
    "bcrypt": "^5.0.1",
    "class-transformer": "^0.4.0",
    "class-validator": "^0.13.1",
    "compression": "^1.7.4",
    "expo-server-sdk": "^3.6.0",
    "generate-password": "^1.6.0",
    "lodash": "^4.17.20",
    "moment": "^2.29.1",
    "mysql2": "^2.2.5",
    "nanoid": "^3.1.22",
    "nestjs-i18n": "^8.1.0",
    "nestjs-typeorm-paginate": "^2.3.2",
    "nodemailer": "^6.5.0",
    "passport-jwt": "^4.0.0",
    "passport-local": "^1.0.0",
    "redis": "^3.1.2",
    "reflect-metadata": "^0.1.13",
    "rimraf": "^3.0.2",
    "rxjs": "^7.4.0",
    "socket.io-redis": "^6.1.1",
    "swagger-ui-express": "^4.1.6",
    "typeorm": "^0.2.31",
    "typeorm-naming-strategies": "^2.0.0",
    "uuid": "^8.3.2"
  },
  "devDependencies": {
    "@nestjs/cli": "^8.1.5",
    "@nestjs/schematics": "^8.0.5",
    "@nestjs/testing": "^8.2.2",
    "@types/aws-sdk": "^2.7.0",
    "@types/bcrypt": "^3.0.0",
    "@types/cron": "^1.7.3",
    "@types/express": "^4.17.8",
    "@types/jest": "^26.0.15",
    "@types/lodash": "^4.14.168",
    "@types/multer": "^1.4.5",
    "@types/node": "^14.14.6",
    "@types/nodemailer": "^6.4.0",
    "@types/passport-jwt": "^3.0.5",
    "@types/passport-local": "^1.0.33",
    "@types/supertest": "^2.0.10",
    "@typescript-eslint/eslint-plugin": "^4.6.1",
    "@typescript-eslint/parser": "^4.6.1",
    "eslint": "^7.12.1",
    "eslint-config-prettier": "7.2.0",
    "eslint-plugin-prettier": "^3.1.4",
    "jest": "^26.6.3",
    "prettier": "^2.1.2",
    "supertest": "^6.0.0",
    "ts-jest": "^26.4.3",
    "ts-loader": "^8.0.8",
    "ts-node": "^9.0.0",
    "tsconfig-paths": "^3.9.0",
    "typescript": "^4.0.5"
  },
````

### Node.js version

v15.13.0

### In which operating systems have you tested?

- [ ] macOS
- [ ] Windows
- [X] Linux

### Other

On the Client Side, I am currently using the `"socket.io-client": "^4.3.2"`.
I also tried the `latest` version (`^4.4.0`) , but it don’t work.

**Comment by jmcdo29** [comment]
Can you provide a minimum git repo that's not a part of the nest main repository? Just makes it easier to test

**Comment by emmanuel-D** [comment]
I finally fixed the issue.
The problem was due to `transports` property of the `@WebSocketGateway` annotation.

To fix it, I explicitly configure the annotation to use the `websocket` as transports like this:
````
@WebSocketGateway({ transports: ['websocket'] })
````
And I also configured the **Client** to use that `transports` type like this:
````
io(url, { transports: ['websocket', 'polling']});
````

I think, when not specified, it can use `polling` or what else.
